pkg/networkservice/common/refresh/client_test.go

```diff
@@ -267,3 +268,24 @@ func TestRefreshClient_Sandbox(t *testing.T) {
 		return refreshSrv.getState() == testRefreshStateDoneRequest
 	}, sandboxTotalTimeout, sandboxStepDuration)
 }
+
+func TestRefreshClient_NoRefreshOnFailure(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client := chain.NewNetworkServiceClient(
+		serialize.NewClient(),
+		updatepath.NewClient("refresh"),
+		refresh.NewClient(ctx),
+		injecterror.NewClient(),
+	)
+
+	_, err := client.Request(ctx, &networkservice.NetworkServiceRequest{
+		Connection: new(networkservice.Connection),
+	})
+	require.Error(t, err)
+
+	goleak.VerifyNone(t)
```

Comment:
Why do we need this?
```suggestion
```

Reply:
To validate that there is no pending refresh action.

Reply:
I don't quite follow we already have check on cleanup at line 273

Reply:
@Bolodya1997 Could you please explain this line?

Reply:
Can we use clockmock instead of an additional goleak call?

Reply:
Yep, fixed.